pysindy/pysindy.py

```diff
@@ -519,21 +526,28 @@ def print(self, lhs=None, precision=3):
             Precision to be used when printing out model coefficients.
         """
         eqns = self.equations(precision)
-        if isinstance(self.optimizer, SINDyPI):
-            feature_names = self.get_feature_names()
+        if sindy_pi_flag:
+            if isinstance(self.optimizer, SINDyPI):
+                feature_names = self.get_feature_names()
+            else:
+                feature_names = self.feature_names
         else:
             feature_names = self.feature_names
         for i, eqn in enumerate(eqns):
             if self.discrete_time:
                 names = "(" + feature_names[i] + ")"
                 print(names + "[k+1] = " + eqn)
             elif lhs is None:
-                if not isinstance(self.optimizer, SINDyPI):
+                if sindy_pi_flag:
+                    if not isinstance(self.optimizer, SINDyPI):
+                        names = "(" + feature_names[i] + ")"
+                        print(names + "' = " + eqn)
+                    else:
+                        names = feature_names[i]
+                        print(names + " = " + eqn)
+                else:
```

Comment:
`if not sindypi_flag or not isinstance(self.optimizer, SINDyPI):`
short-circuits, so if `sindy_pi_flag is False`, python doesn't evaluate the next bit (and thus raise a `NameError` checking for `SINDyPI`).  This lets you combine the nested conditional.

This can be done for the above nested conditional as well.